Sources/SwiftGRPCNIO/HTTP1ToRawGRPCServerCodec.swift

```diff
@@ -0,0 +1,122 @@
+import Foundation
+import NIO
+import NIOHTTP1
+
+public enum RawGRPCServerRequestPart {
+  case head(HTTPRequestHead)
+  case message(ByteBuffer)
+  case end
+}
+
+public enum RawGRPCServerResponsePart {
+  case headers(HTTPHeaders)
+  case message(ByteBuffer)
+  case status(GRPCStatus)
+}
+
+/// A simple channel handler that translates HTTP/1 data types into gRPC packets,
+/// and vice versa.
+public final class HTTP1ToRawGRPCServerCodec: ChannelInboundHandler, ChannelOutboundHandler {
+  public typealias InboundIn = HTTPServerRequestPart
+  public typealias InboundOut = RawGRPCServerRequestPart
+
+  public typealias OutboundIn = RawGRPCServerResponsePart
+  public typealias OutboundOut = HTTPServerResponsePart
+
+  enum State {
+    case expectingHeaders
+    case expectingCompressedFlag
+    case expectingMessageLength
+    case receivedMessageLength(UInt32)
+
+    var expectingBody: Bool {
+      switch self {
+      case .expectingHeaders: return false
+      case .expectingCompressedFlag, .expectingMessageLength, .receivedMessageLength: return true
+      }
+    }
+  }
+
+  private(set) var state = State.expectingHeaders
+
+  private(set) var buffer: NIO.ByteBuffer?
+
+  public func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+    switch self.unwrapInboundIn(data) {
+    case .head(let requestHead):
+      guard case .expectingHeaders = state
+        else { preconditionFailure("received headers while in state \(state)") }
+
+      state = .expectingCompressedFlag
+      buffer = ctx.channel.allocator.buffer(capacity: 5)
+
+      ctx.fireChannelRead(self.wrapInboundOut(.head(requestHead)))
+
+    case .body(var body):
+      guard var buffer = buffer
+        else { preconditionFailure("buffer not initialized") }
+      assert(state.expectingBody, "received body while in state \(state)")
+      buffer.write(buffer: &body)
+
+      // Iterate over all available incoming data, trying to read length-delimited messages.
+      // Each message has the following format:
+      // - 1 byte "compressed" flag (currently always zero, as we do not support for compression)
+      // - 4 byte signed-integer payload length (N)
+      // - N bytes payload (normally a valid wire-format protocol buffer)
+      requestProcessing: while true {
+        switch state {
+        case .expectingHeaders: preconditionFailure("unexpected state \(state)")
+        case .expectingCompressedFlag:
+          guard let compressionFlag: Int8 = buffer.readInteger() else { break requestProcessing }
+          precondition(compressionFlag == 0, "unexpected compression flag \(compressionFlag)")
```

Comment:
Might want to note that this is only unexpected because SwiftGRPC doesn't yet support compression flags

Reply:
It's also unexpected because we do not indicate that we accept compression. Extended the message and added a FIXME.